### ISSUE TITLE
False warning RCS1012 with method Parse

### ISSUE THREAD

**Issue by NN---** [JosefPihrt/Roslynator#748]
**Product and Version Used**: 
3.0.0

**Steps to Reproduce**:
editorconfig
```
dotnet_diagnostic.rcs1012.severity=warning
```

a.cs
```cs
var ip = IPAddress.Parse("8.8.8.8");
```

From Roslyn point of view the type is evident.

**Comment by JosefPihrt** [comment]
I know what you mean.

Question is whether it is good idea. Currently the rule is clear: RHS is either a literal (for example `false`) or it contains actual type (for example `new Foo()`.

I'm afraid that by adding more cases the analyzer would became less clear.

**Comment by NN---** [comment]
The problem is difference between Roslyn and Roslynator.

**Comment by JosefPihrt** [comment]
I'm not sure what do you mean by "Roslyn". It is too generic term. Do you mean built-in analyzers?

**Comment by NN---** [comment]
With .editorconfig:

```
root=true
[*]
csharp_style_var_elsewhere=false:suggestion
csharp_style_var_for_built_in_types=false:suggestion
csharp_style_var_when_type_is_apparent=true:suggestion
```

In the code

```
using System.Net;

public class C
{
    public static void Main()
    {
        IPAddress ip = IPAddress.Parse("a");
    }
}
```

You have an informational message in Visual Studio:

```
Severity	Code	Description	Project	File	Line	Suppression State
Message	IDE0007	use 'var' instead of explicit type	ConsoleApp	Program.cs	7	Active
```

**Comment by JosefPihrt** [comment]
Thanks for detailed comment.

I think it reasonable to make this rule consistent with Microsoft built-in rule.
